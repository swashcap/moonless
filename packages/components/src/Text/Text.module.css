/** Text */

.Text {
  margin-bottom: 0;
  margin-top: 0;
}

/** Text: Title */

.TextTitle {
  font-size: var(--typography-font-size-600);
  font-weight: var(--typography-font-weight-normal);
  line-height: var(--typography-line-height-600);
}

/** Text: Headline 1 */

.TextHeadline1 {
  font-size: var(--typography-font-size-500);
  font-weight: var(--typography-font-weight-normal);
  line-height: var(--typography-line-height-500);
}

/** Text: Headline 2 */

.TextHeadline2 {
  font-size: var(--typography-font-size-400);
  font-weight: var(--typography-font-weight-normal);
  line-height: var(--typography-line-height-400);
}

/** Text: Headline 3 */

.TextHeadline3 {
  font-size: var(--typography-font-size-300);
  font-weight: var(--typography-font-weight-normal);
  line-height: var(--typography-line-height-300);
}

/** Text: Subhead 1 */

.TextSubhead1 {
  font-size: var(--typography-font-size-200);
  font-weight: var(--typography-font-weight-bold);
  line-height: var(--typography-line-height-200);
}

/** Text: Subhead 2 */

.TextSubhead2 {
  font-size: var(--typography-font-size-100);
  font-weight: var(--typography-font-weight-bold);
  line-height: var(--typography-line-height-100);
}

/** Text: Body */

.TextBody {
  font-size: var(--typography-font-size-200);
  font-weight: var(--typography-font-weight-normal);
  line-height: var(--typography-line-height-200);
}

/** Text: Caption */

.TextCaption {
  font-size: var(--typography-font-size-100);
  font-weight: var(--typography-font-weight-normal);
  line-height: var(--typography-line-height-100);
}
