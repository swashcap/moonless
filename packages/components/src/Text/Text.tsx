import { clsx } from 'moonless-utils';

import styles from './Text.module.css';

export type TextVariant =
  | 'title'
  | 'headline-1'
  | 'headline-2'
  | 'headline-3'
  | 'subhead-1'
  | 'subhead-2'
  | 'body'
  | 'caption';

export type TextElement =
  | 'h1'
  | 'h2'
  | 'h3'
  | 'h4'
  | 'h5'
  | 'h6'
  | 'p'
  | 'small'
  | 'span';

export type TextProps<
  T extends TextElement
> = React.HTMLAttributes<HTMLElement> & {
  as?: T;
  /**
   * The text's size/type variation.
   *
   * @default 'body'
   */
  variant?: TextVariant;
};

export const Text = <T extends TextElement = 'span'>({
  as,
  className,
  variant,
  ...rest
}: TextProps<T>) => {
  const Component = as ?? 'span';

  return (
    <Component
      className={clsx(
        styles.Text,
        variant === 'body' && styles.TextBody,
        variant === 'caption' && styles.TextCaption,
        variant === 'headline-1' && styles.TextHeadline1,
        variant === 'headline-2' && styles.TextHeadline2,
        variant === 'headline-3' && styles.TextHeadline3,
        variant === 'subhead-1' && styles.TextSubhead1,
        variant === 'subhead-2' && styles.TextSubhead2,
        variant === 'title' && styles.TextTitle,
        className
      )}
      {...rest}
    />
  );
};
