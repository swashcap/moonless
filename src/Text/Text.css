.Text {
  font-size: 100%;
  line-height: inherit;
}
