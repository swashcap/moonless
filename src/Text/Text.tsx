import React from 'react';

import { clsx } from '../utils/clsx';

export type TextProps = React.DetailedHTMLProps<
  React.HTMLAttributes<HTMLSpanElement>,
  HTMLSpanElement
>;

export const Text: React.FC<TextProps> = ({ className, ...rest }) => (
  <span className={clsx('Text', className)} {...rest} />
);
