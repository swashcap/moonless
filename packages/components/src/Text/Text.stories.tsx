import React from 'react';

import { Text } from './Text';

export default {
  title: 'Component/Text',
};

export const Default = () => <Text>Text</Text>;

export const As = () => (
  <>
    <Text as="h1">h1</Text>
    <Text as="h2">h2</Text>
    <Text as="p">p</Text>
    <Text as="small">small</Text>
  </>
);

export const Variant = () => (
  <>
    <Text as="h1" variant="title">
      title
    </Text>
    <Text as="h2" variant="headline-1">
      headline-1
    </Text>
    <Text as="h3" variant="headline-2">
      headline-2
    </Text>
    <Text as="h4" variant="headline-3">
      headline-3
    </Text>
    <Text as="h5" variant="subhead-1">
      subhead-1
    </Text>
    <Text as="h6" variant="subhead-2">
      subhead-2
    </Text>
    <Text as="p" variant="body">
      body
    </Text>
    <Text as="small" variant="caption">
      caption
    </Text>
  </>
);
