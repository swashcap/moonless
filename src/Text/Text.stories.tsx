import React from 'react';

import { Text as TextComponent } from './Text';

export default {
  title: 'Component|Text',
};

export const Text = () => <TextComponent>Text</TextComponent>;
