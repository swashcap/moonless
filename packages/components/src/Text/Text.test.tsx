import React from 'react';
import { render } from '@testing-library/react';

import { Text } from './Text';

describe('Text', () => {
  test('It should render.', () => {
    const { container } = render(<Text>Text</Text>);

    expect(container).toMatchSnapshot();
  });
});
