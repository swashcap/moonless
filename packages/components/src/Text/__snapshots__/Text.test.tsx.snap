// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text It should render. 1`] = `
<div>
  <span
    class="Text"
  >
    Text
  </span>
</div>
`;
